src/custom/pages/Swap/ArrowWrapperLoader.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import loadingCowGif from 'assets/cow-swap/cow-load.gif'
+import useLoadingWithTimeout from 'hooks/useLoadingWithTimeout'
+import { useIsQuoteRefreshing } from 'state/price/hooks'
+
+export interface ArrowWrapperProps {
+  children: React.ReactNode
+}
+
+export function ArrowWrapperLoader({ children }: ArrowWrapperProps) {
+  const COW_LOADING_TIME = 4000
+  const isRefreshingQuote = useIsQuoteRefreshing()
+  const showLoader = useLoadingWithTimeout(isRefreshingQuote, COW_LOADING_TIME)
+
+  return (
+    <Wrapper showLoader={showLoader}>
+      {children}
+      <div>{showLoader && <img src={loadingCowGif} alt="Loading prices..." />}</div>
+    </Wrapper>
+  )
+}
+
+export const Wrapper = styled.div<{ showLoader: boolean }>`
+  position: absolute;
+  background: ${({ theme }) => theme.swap.arrowDown.background};
+  border-radius: ${({ theme }) => theme.swap.arrowDown.borderRadius};
+  width: ${({ theme }) => theme.swap.arrowDown.width};
+  height: ${({ theme }) => theme.swap.arrowDown.height};
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border: ${({ theme }) => `${theme.swap.arrowDown.borderSize} solid ${theme.swap.arrowDown.borderColor}`};
+  transition: transform 0.1s ease-in-out;
```

Comment:
not a fan of grouping css styles like this. easier to read when similar properties are grouped together, helps avoid mistakes like the 2 transition properties you have in this CSS

Reply:
Can review and cleanup 👍🏼